lib/_stream_readable.js

```diff
@@ -1070,13 +1072,90 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
-  if (createReadableStreamAsyncIterator === undefined) {
-    createReadableStreamAsyncIterator =
-      require('internal/streams/async_iterator');
+  let stream = this;
+
+  if (typeof stream.read !== 'function') {
+    // v1 stream
+    const src = stream;
+    stream = new Readable({
+      objectMode: true,
+      destroy(err, callback) {
+        destroyImpl.destroyer(src, err);
+        callback();
+      }
+    }).wrap(src);
   }
-  return createReadableStreamAsyncIterator(this);
+
+  const iter = createAsyncIterator(stream);
+  iter.stream = stream;
+  return iter;
 };
 
+async function* createAsyncIterator(stream) {
```

Comment:
While we are changing the implementation of our async iterator, we should enforce that creating more than one async iterator for a given stream is an error. I think it should `throw`.

Reply:
If you don't mind I think we should do this as a separate discussion & PR. While I do agree with this approach I would like us to further consider it, e.g. as @benjamingr points out above, it actually might work and be a valid use case to have multiple for awaits on the same stream (i.e. multiple iterators), not sure if it is a good idea though, the problem is using the same iterator in multiple for awaits, which I don't see how we could detect.

Reply:
>  we should enforce that creating more than one async iterator for a given stream is an error.

See the discussion in https://github.com/nodejs/node/pull/34035#issuecomment-650226447 , I think this already probably works.